src/Microsoft.AspNetCore.Razor.Language/Components/CodeGenerationConstants.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    // Constants for method names used in code-generation
+    // Keep these in sync with the actual definitions
+    internal static class CodeGenerationConstants
+    {
+        public static class RazorComponent
+        {
+            public const string FullTypeName = "Microsoft.AspNetCore.Components.Component";
+            public const string BuildRenderTree = nameof(BuildRenderTree);
```

Comment:
ewww `nameof` is wrong here. Renaming this constant should **not** change the codegen

Reply:
I copied this from Blazor - https://github.com/aspnet/Blazor/blob/master/src/shared/BlazorApi.cs#L18

Reply:
It's still wrong.